### PR TITLE
ref(teamKeyTransaction): Use CompactSelect

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before ——**
Screenshot 2023-06-12 at 6 16 11 PM
Screenshot 2023-06-12 at 6 17 45 PM

**After ——**
Screenshot 2023-06-12 at 6 15 45 PM
Screenshot 2023-06-12 at 6 17 20 PM

### CODE REVIEW

File: static/app/utils/discover/fieldRenderers.tsx [hunk 1]
```diff
@@ -624,14 +624,12 @@ const SPECIAL_FIELDS: SpecialFields = {
...
-          projectSlug={data.project}
-          transactionName={data.transaction}
-        />
-      </Container>
+      <TeamKeyTransactionField
+        isKeyTransaction={(data.team_key_transaction ?? 0) !== 0}
+        organization={organization}
+        projectSlug={data.project}
+        transactionName={data.transaction}
+      />
```
Comment (vuluongj20) [inline static/app/utils/discover/fieldRenderers.tsx:632]: `Container` only adds overflow ellipsis styles, which aren't necessary here.

File: static/app/components/performance/teamKeyTransaction.tsx [hunk 2]
```diff
@@ -1,416 +1,132 @@
...
 
+  const options = useMemo<MultipleSelectProps<string>['options']>(() => {
     const enabledTeams: Team[] = [];
     const disabledTeams: Team[] = [];
-    const noAccessTeams: Team[] = [];
 
-    const projectTeams = new Set(project.teams.map(({id}) => id));
     for (const team of teams) {
       if (!projectTeams.has(team.id)) {
-        noAccessTeams.push(team);
```
Comment (vuluongj20) [inline static/app/components/performance/teamKeyTransaction.tsx:101]: @doralchan and I both agree that we don't need to include "no-access" teams in the menu
image

Despite its name, the "no-access" section doesn't contain teams that the user doesn't have access to. Rather, it's the list of teams that aren't associated with the project that the highlighted transaction is in.

For example, the following transaction is in the `javascript` project. That project is associated with the `sentry` and `support-test` teams. The `new-page-filters` team is not associated with `javascript`, so it's in the "no-access" section.
image

There doesn't seem to be a good reason for `new-page-filters` to be included. If it was a matter of user access, then displaying the team alongside a link to request access would make sense. But that's not the case here.

File: static/app/components/performance/teamKeyTransaction.tsx [hunk 3]
```diff
@@ -1,416 +1,132 @@
-import {Component, Fragment} from 'react';
-import {createPortal} from 'react-dom';
-import {Manager, Popper, Reference} from 'react-popper';
```
Comment (vuluongj20) [inline static/app/components/performance/teamKeyTransaction.tsx:3]: @evanpurkhiser Another one 😛

**Review by scttcper — APPROVED** [review]
i'll approve it 🤷

**Pull request merged**